libs/libcommon/src/libcommon/queue.py

```diff
@@ -198,6 +202,53 @@ def flat_info(self) -> FlatJobInfo:
         )
 
 
+class Lock(Document):
+    meta = {"collection": QUEUE_COLLECTION_LOCKS, "db_alias": QUEUE_MONGOENGINE_ALIAS}
+    key = StringField(primary_key=True)
+    job_id = StringField()
+
+    created_at = DateTimeField(required=True)
+    updated_at = DateTimeField()
+
```

Comment:
you will have to create some indexes to speed up the queries

Reply:
I don't think it's needed since I use the primary key in the queries, and there's always an index for the primary key

Reply:
hmmm, for https://github.com/huggingface/datasets-server/pull/1357/files#diff-44fa7cb2645881e55953db64dafa198b2e007a2e531f70acaeebfc50ffa67953R228, I think we need 

```
"indexes": [
            ("key", "job_id")
]
```